File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
import math

def solution(n,a,b):
    match_count = math.log(n, 2)
    while match_count > 1:
        n /= 2
        if (a > n and b <= n) or (a <= n and b > n):
            return match_count
        if a - n > 0:
            a -= n
            b -= n
        match_count -= 1
    return match_count

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int roundNum = 0;
        for(int i = 1; n > i; i*=2) {
            roundNum++;
        }

        for (int i = roundNum; i > 0; i--) {
            int middleNum = (int)Math.pow(2, i-1);
            if(a > middleNum) {
                if(b <= middleNum) return i;
                a -= middleNum;
                b -= middleNum;
            }
            else if(b > middleNum) {
                if(a <= middleNum) return i;
                a -= middleNum;
                b -= middleNum;
            }
        }

        return 0;
    }
}